codegen/src/main/scala/bsp/codegen/Lines.scala

```diff
@@ -0,0 +1,59 @@
+package bsp.codegen
+
+import cats.kernel.Monoid
+
+trait Lines { self =>
```

Comment:
This is a construct that's convenient to manipulate for indentation and stuff. We use a similar thing in smithy4s, though more complex.

Reply:
Just for clarity, can you add some scaladoc explaining this, especially since there are other things related to it like the `ToLines` and the implicits there. I want to solidify for future people why we have this.